### PR TITLE
Feature/597 upload and transform documents

### PR DESCRIPTION
**Pull request by neoxelox**
Related to: https://github.com/latitude-dev/latitude-llm/issues/597

- Allow to upload documents to the project and transform them as a promptl document in raw text

### CODE REVIEW

File: packages/core/src/services/documents/create.ts [hunk 1]
```diff
@@ -39,6 +40,14 @@ export async function createNewDocument(
       return Result.error(new BadRequestError('Cannot modify a merged commit'))
     }
 
+    if (!DOCUMENT_PATH_REGEXP.test(path)) {
```
Comment (andresgutgon) [inline packages/core/src/services/documents/create.ts:43]: Do  this in the action in the zod validator

File: packages/core/src/services/documents/create.ts [hunk 2]
```diff
@@ -51,27 +60,14 @@ export async function createNewDocument(
...
-    if (provider) metadata += `provider: ${provider.name}`
-
-    const model = findFirstModelForProvider({
-      provider: provider,
-      latitudeProvider: env.DEFAULT_PROVIDER_ID,
-    })
-    if (model) metadata += `\nmodel: ${model}`
-
-    const defaultContent = metadata ? `---\n${metadata}\n---` : ''
+    const def = await defaultDocumentContent({ workspace }, tx)
```
Comment (andresgutgon) [inline packages/core/src/services/documents/create.ts:63]: use a better variable name

File: packages/core/src/services/files/transform.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+          putNotesAtLast: false,
+        })
+        break
+      default:
+        if (!isText(file.name, buffer)) {
+          return Result.error(
+            new BadRequestError(`Unsupported file type: ${extension}`),
+          )
+        }
+        content = buffer.toString()
```
Comment (andresgutgon) [inline packages/core/src/services/files/transform.ts:56]: So a `.md` or `.text` will arrive here no?

File: packages/web-ui/src/sections/Document/Sidebar/Files/TreeToolbar/index.tsx [hunk 4]
```diff
@@ -43,25 +51,83 @@ export function TreeToolbar() {
...
+                lookDisabled={isMerged}
+                disabled={isLoading}
+                iconProps={{ name: 'filePlus' }}
+                onClick={onClick(EntityType.File)}
+              />
+            }
+          >
+            New prompt
+          </Tooltip>
+          <div>
```
Comment (andresgutgon) [inline packages/web-ui/src/sections/Document/Sidebar/Files/TreeToolbar/index.tsx:107]: Why this div?

**Review by andresgutgon — DISMISSED** [review]

File: packages/core/src/services/files/transform.ts (hunk 3 above)
Comment (neoxelox) [reply packages/core/src/services/files/transform.ts:56]: yes, any file that is raw text

File: packages/web-ui/src/sections/Document/Sidebar/Files/index.tsx [hunk 5]
```diff
@@ -191,13 +195,17 @@ export function FilesTree({
...
+        isLoading={isLoading}
         isMerged={isMerged}
         onMergeCommitClick={onMergeCommitClick}
         currentUuid={currentUuid}
         onNavigateToDocument={navigateToDocument}
         onCreateFile={(path) => {
           createFile({ path })
         }}
+        onUploadFile={({ path, file }) => {
+          uploadFile({ path, file })
```
[the remark this answers is not in the document]
Comment (neoxelox) [reply packages/web-ui/src/sections/Document/Sidebar/Files/index.tsx:207]: No need but did it for consistency

File: apps/web/package.json [hunk 6]
```diff
@@ -10,7 +10,6 @@
     "dev:debug": "NODE_DEBUG=latitude:debug NODE_OPTIONS='--inspect=0.0.0.0:9229' ./node_modules/.bin/next dev",
     "dev:local": "USE_LOCALHOST=true pnpm run dev",
     "lint": "next lint",
-    "prettier": "prettier --write \"**/*.{ts,tsx,md}\"",
```
Comment (geclos) [inline apps/web/package.json:13]: hey

File: apps/gateway/package.json [hunk 7]
```diff
@@ -9,7 +9,6 @@
     "build": "tsup --config tsup.config.ts && pnpm run sentry:sourcemaps",
     "dev:debug": "tsx watch --inspect src/server",
     "lint": "eslint src/",
-    "prettier": "prettier --write \"**/*.{ts,tsx,md}\"",
```
Comment (geclos) [inline apps/gateway/package.json:12]: why?

**Pull request merged**